main.py

```diff
@@ -1,382 +1,233 @@
+        self.tts.play()
+        return ""
 
+    def pause(self):
+        self.tts.pause()
+        return ""
 
-@app.route("/skip")
-def skip():
-    global paused
-    paused = False
-    if play_process is not None:
-        os.kill(play_process.pid, signal.SIGCONT)
-        os.killpg(play_process.pid, signal.SIGTERM)
-        return "Skipped current playing"
-    return "Nothing playing"
+    def reset(self):
+        self.tts.reset()
+        return ""
 
+    def skip(self):
+        self.tts.skip()
+        return ""
 
-def notify(msg):
-    global notifier
-    notifier.send_sync(title="TTS Reader", message=msg, timeout=2)
+    def speed(self, data):
+        data = max(0.0, min(data, 10.0))
+        self.parsed.speed = data
+        return ""
 
+    def volume(self, data):
+        data = max(0.0, min(data, 1.0))
+        self.parsed.volume = data
+        return ""
 
-def uptime():
-    diff = time.time() - begin_time
-    return str(datetime.timedelta(seconds=int(diff)))
+    def uptime(self):
+        diff = time.time() - self.begin_time
+        return str(datetime.timedelta(seconds=int(diff)))
 
+    def run(self):
+        self.flask.run(
+            host=self.parsed.ip, port=self.parsed.port, debug=self.parsed.debug
+        )
 
-def fatal_exit():
-    os.kill(os.getpid(), signal.SIGTERM)
+    def notify(self, msg):
+        self.notifier.send_sync(title="TTS Reader", message=msg, timeout=2)
 
 
 if __name__ == "__main__":
-    begin_time = time.time()
-    parsed = parser.parse_args()
+    parser = argparse.ArgumentParser(
+        prog="tts-reader",
+    )
+    parser.add_argument("--ip", type=str, default="127.0.0.1", help="IP address")
+    parser.add_argument("--port", type=int, default=5000, help="Port")
+    parser.add_argument(
+        "--wayland",
+        default=False,
+        action=argparse.BooleanOptionalAction,
+        help="Assume running under Wayland",
+    )
+    parser.add_argument(
+        "--piper-python",
+        default=False,
+        action=argparse.BooleanOptionalAction,
+        help="Attempt to use the piper python module. Has no effect if a different backend is selected",
+    )
+    parser.add_argument(
+        "--speechd",
+        default=False,
+        action=argparse.BooleanOptionalAction,
+        help="Use speechd instead of piper. Incomplete",
+    )
+    parser.add_argument("--volume", type=float, default=1.0, help="Volume [0-1]")
+    parser.add_argument(
+        "--speed", type=float, default=1.0, help="Playback speed [0-10]"
+    )
+    parser.add_argument(
+        "--piper-rate",
+        type=int,
+        default=22050,
+        help="Piper: Playback sample rate. More info at https://github.com/rhasspy/piper/blob/master/TRAINING.md",
+    )
+    parser.add_argument(
+        "--piper-sentence-silence",
+        type=float,
+        default=0.8,
+        help="Piper: Seconds of silence after each sentence",
+    )
+    parser.add_argument(
+        "--piper-one-sentence",
```

Comment:
I think you might have missed implementing the one-sentence part. You could add this if you want or I should be able to add this during the weekend.

Reply:
Yeah. I can do it